src/test/java/org/shikshalokam/backend/scp/TestScpEntityTypesAndEntitiesCRUDOperations.java

```diff
@@ -0,0 +1,239 @@
+        Assert.assertTrue(statusCode == 400 || statusCode == 422, "Status code should be 400 or 422 for invalid payload");
+
+        logger.info("Ended calling the entity types API with valid payload.");
+    }
+
+    @Test(dependsOnMethods = "testCreateEntityTypeWithEmptyFields", description = "Verifies the functionality of updating user's permission with Invalid payload.")
+    public void testUpdateEntityTypesWithEmptyFields() {
+        logger.info("Started calling the entity types API with empty fields payload:");
+
+        // Call entityTypeUpdation with invalid parameters
+        JSONObject requestBody = entityTypeForUpdate("", "", "", "", "", "");
+
+        // Call entityTypeUpdation with the created requestBody
+        Response response = entityTypeUpdateRequest(requestBody);
+
+        // Log the status code and response body
+        int statusCode = response.getStatusCode();
+        response.prettyPrint();
+
+        // Validate response code for empty fields (usually 400)
+        Assert.assertEquals(statusCode, 400, "Status code should be 400 for empty fields payload");
+
+        logger.info("Ended calling the entity types API with empty fields.");
+    }
+
+    //Method to create request body for entityTypes
+    private JSONObject createEntityType(String value, String label, String status, String type, String data_type, String has_entities) {
+        JSONObject requestBody = new JSONObject();
+
+        // Generate random alphabetic string for "code"
+        requestBody.put("value", value + RandomStringUtils.randomAlphabetic(8).toLowerCase());
+        requestBody.put("label", label + RandomStringUtils.randomAlphabetic(8).toLowerCase());
+        requestBody.put("status", status);
+        requestBody.put("type", type);
+        requestBody.put("data_type", data_type);
+        requestBody.put("has_entities", has_entities);
+
+        // Pretty-print the response for debugging
+        response.prettyPrint();
+
+        return requestBody;
+    }
+
+    private Response createEntityTypeRequest(JSONObject requestBody) {
+        try {
+            entityTypeCreateEndpoint = new URI(PROP_LIST.get("scp.create.entitytype.endpoint").toString());
+        } catch (URISyntaxException e) {
+            throw new RuntimeException("Invalid URI for entityTypeCreatEndpoint", e);
+        }
+
+        // Make the POST request to update the permission
+        Response response = given()
+                .header("X-auth-token", "bearer " + X_AUTH_TOKEN)
+                .contentType(ContentType.JSON)
+                .body(requestBody.toString())
+                .when().post(entityTypeCreateEndpoint);
+
+        // Pretty-print the response for debugging
+        response.prettyPrint();
+
+        return response;
+    }
+
+    // Method to create request body for update entityTypes
+    private JSONObject entityTypeForUpdate(String value, String label, String status, String type, String allow_filtering, String data_type) {
```

Comment:
you can do this way so that method will take single argument and code will reduce 
private JSONObject entityTypeForUpdate ( HasMap<String,String > object )
{
JSONObject jsonObject = new JSONObject(map);
}